packages/vaadin-grid/test/selection.test.js

```diff
@@ -392,6 +392,16 @@ describe('multi selection column', () => {
     expect(selectionColumn.selectAll).to.be.false;
   });
 
+  it('should not clear pre-selected items', () => {
+    const grid = fixtureSync(`
+      <vaadin-grid selected-items='[{"value": "foo"}]'>
+        <vaadin-grid-selection-column></vaadin-grid-selection-column>
+        <vaadin-grid-column path="value"></vaadin-grid-column>
+      </vaadin-grid>
+    `);
+    expect(grid.selectedItems).to.eql([{ value: 'foo' }]);
```

Comment:
Probably, you missed adding `deep` to enable deep equality mode here. Otherwise, it might be better to check the count of selected items.

Reply:
> Probably, you missed adding deep to enable deep equality mode here

Note that `eql` does a deep equality check.

But as you suggested, it's cleaner to check for the array length instead.

Reply:
Oh, then I'm sorry, I didn't know that :)